Modules/ZDC/src/ZDCRecDataCheck.cxx

```diff
@@ -0,0 +1,321 @@
+        mQADC = 2;
+      }
+      if (mNumEADC > 0) {
+        mQADC = 3;
+      }
+    }
+    mNumETDC = 0;
+    mNumWTDC = 0;
+    mStringWTDC = "";
+    mStringETDC = "";
+    if (mo->getName() == "h_summmary_TDC") {
+      auto* h = dynamic_cast<TH1F*>(mo->getObject());
+      // dumpVecParam((int)h->GetNbinsX(),(int)mVectParamTDC.size());
+      if ((int)h->GetNbinsX() != (int)mVectParamTDC.size())
+        return Quality::Null;
+      for (int i = 0; i < h->GetNbinsX(); i++) {
+        ib = i + 1;
+        if ((((float)h->GetBinContent(ib) < (float)mVectParamTDC.at(i).minW && (float)h->GetBinContent(ib) >= (float)mVectParamTDC.at(i).minE)) || ((float)h->GetBinContent(ib) > (float)mVectParamTDC.at(i).maxW && (float)h->GetBinContent(ib) < (float)mVectParamTDC.at(i).maxE)) {
+          mNumWTDC += 1;
+          mStringWTDC = mStringWTDC + mVectParamTDC.at(i).ch + " ";
+          ILOG(Warning, Support) << "Rec Warning in " << mVectParamTDC.at(i).ch << " intervall: " << mVectParamTDC.at(i).minW << " - " << mVectParamTDC.at(i).maxW << " Value: " << h->GetBinContent(ib) << ENDM;
+        }
+        if (((float)h->GetBinContent(ib) < (float)mVectParamTDC.at(i).minE) || ((float)h->GetBinContent(ib) > (float)mVectParamTDC.at(i).maxE)) {
+          mNumETDC += 1;
+          mStringETDC = mStringETDC + mVectParamTDC.at(i).ch + " ";
+          ILOG(Error, Support) << "Rec Error in " << mVectParamTDC.at(i).ch << " intervall: " << mVectParamTDC.at(i).minE << " - " << mVectParamTDC.at(i).maxE << " Value: " << h->GetBinContent(ib) << ENDM;
+        }
+      }
+      if (mNumWTDC == 0 && mNumETDC == 0) {
+        mQTDC = 1;
+      }
+      if (mNumWTDC > 0) {
+        mQTDC = 2;
+      }
+      if (mNumETDC > 0) {
+        mQTDC = 3;
+      }
+
+      if (mQADC == 1 && mQTDC == 1) {
+        result = Quality::Good;
+        result.addReason(FlagReasonFactory::Unknown(), "ZDCRecDataTask quality is Good");
+      } else if (mQADC == 3 || mQTDC == 3) {
+        result = Quality::Bad;
+        if (mQADC == 3)
+          result.addReason(FlagReasonFactory::Unknown(),
+                           "Task quality is bad because in ADC Summary " + std::to_string(mNumWADC) + " channels:" + mStringEADC + "have a value in the bad range");
+        if (mQTDC == 3)
+          result.addReason(FlagReasonFactory::Unknown(),
+                           "It is bad because in TDC Summary" + std::to_string(mNumWTDC) + " channels:" + mStringWTDC + "have a value in the bad range");
+        if (mQADC == 2)
+          result.addReason(FlagReasonFactory::Unknown(),
+                           "It is medium because in ADC Summary " + std::to_string(mNumWADC) + " channels:" + mStringWADC + "have a value in the medium range");
+
+        if (mQTDC == 2)
+          result.addReason(FlagReasonFactory::Unknown(),
+                           "It is medium because in TDC Summary " + std::to_string(mNumWTDC) + " channels:" + mStringWTDC + "have a value in the medium range");
```

Comment:
here you are saying it is medium, but you are setting the quality::bad, is that a typo?